File: sort/bubble_sort/go/bubble_sort.go
package bubblesort

// BubbleSort uses the bubble sort algorithm to sort an integer slice.
func BubbleSort(arr []int) {
	sorted := false
	for !sorted {
		sorted = true
		for i := range arr[:len(arr)-1] {
			if arr[i] > arr[i+1] {
				arr[i], arr[i+1] = arr[i+1], arr[i]
				sorted = false
			}
		}
	}
}

File: sort/bubble_sort/Rust/bubble_sort.rs
fn bubble_sort(mut arr :Vec<i32>) -> Vec<i32> {
    for _ in 0..arr.len() {
        let mut flag = false;
        for j in 0..arr.len() - 1 {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j+1);
                flag = true
            }
        }
        if !flag {
            break;
        }
    }
    arr
}

fn main() {
    let arr = vec![2, 1, 2,7,88,4,3,6];
    println!("Sorted array: {:?}", bubble_sort(arr));
}